From 15ecaeab73b3b8495c7bd8a98ce0da250a2079d9 Mon Sep 17 00:00:00 2001
From: Carlos O'Ryan <coryan@google.com>
Date: Fri, 13 Oct 2023 06:18:18 -0400
Subject: [PATCH 2/2] fix(storage): remove unused dependencies (#12890)

We do not directly use `OpenSSL::SSL` or `ZLIB::ZLIB`. Both are used
indirectly via `CURL::libcurl`. We should not require the package or
link the libraries.
---
 google/cloud/storage/google_cloud_cpp_storage.cmake      | 5 +----
 google/cloud/storage/google_cloud_cpp_storage_grpc.cmake | 4 +---
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/google/cloud/storage/google_cloud_cpp_storage.cmake b/google/cloud/storage/google_cloud_cpp_storage.cmake
index 92d740a4..629860cd 100644
--- a/google/cloud/storage/google_cloud_cpp_storage.cmake
+++ b/google/cloud/storage/google_cloud_cpp_storage.cmake
@@ -16,7 +16,6 @@
 
 include(FindCurlWithTargets)
 find_package(OpenSSL REQUIRED)
-find_package(ZLIB REQUIRED)
 
 # the client library
 add_library(
@@ -259,9 +258,7 @@ target_link_libraries(
            Crc32c::crc32c
            CURL::libcurl
            Threads::Threads
-           OpenSSL::SSL
-           OpenSSL::Crypto
-           ZLIB::ZLIB)
+           OpenSSL::Crypto)
 if (WIN32)
     # We use `setsockopt()` directly, which requires the ws2_32 (Winsock2 for
     # Windows32?) library on Windows.
diff --git a/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake b/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake
index 5be4813c..8f292fc3 100644
--- a/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake
+++ b/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake
@@ -144,9 +144,7 @@ else ()
                Crc32c::crc32c
                CURL::libcurl
                Threads::Threads
-               OpenSSL::SSL
-               OpenSSL::Crypto
-               ZLIB::ZLIB)
+               OpenSSL::Crypto)
     google_cloud_cpp_add_common_options(google_cloud_cpp_storage_grpc)
     target_include_directories(
         google_cloud_cpp_storage_grpc
-- 
2.42.0.655.g421f12c284-goog

